src/main/java/dev/enjarai/trickster/fleck/FleckType.java

```diff
@@ -0,0 +1,22 @@
+package dev.enjarai.trickster.fleck;
+
+import com.mojang.serialization.Lifecycle;
+import dev.enjarai.trickster.Trickster;
+import io.wispforest.endec.StructEndec;
+import net.minecraft.registry.Registry;
+import net.minecraft.registry.RegistryKey;
+import net.minecraft.registry.SimpleRegistry;
+
+public record FleckType<T extends Fleck>(StructEndec<T> endec) {
+
+    public static final RegistryKey<Registry<FleckType<?>>> REGISTRY_KEY = RegistryKey.ofRegistry(Trickster.id("fleck_type"));
+    public static final Registry<FleckType<?>> REGISTRY = new SimpleRegistry<>(REGISTRY_KEY, Lifecycle.stable());
```

Comment:
Registry should be made in a fashion similar to those of Louis' recently merged PR.

Reply:
made them like the spell executor type. which works i think? it builds.